### PR TITLE
Add content to sphinx documentation

### PR DESCRIPTION
**Pull request by slfuchs**
This PR adds an installation guide to the Sphinx documentation.

Besides, a file structure for the development guide is added. However, the actual content shall be part of a future PR.

### CODE REVIEW

**Comment by slfuchs** [comment]
@ppraegla You can download the generated Sphinx documentation from the artifacts here: https://github.com/PartExa/PartExa/actions/runs/1671743173

File: doc/sphinx/installation/dependencies.rst [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+.. code-block:: console
+
+   git clone https://github.com/dealii/dealii.git <dealii_source>
+
+Download the (latest) p4est release tarball
+
+.. code-block:: console
+
+   cd <p4est_source>
```
Comment (ppraegla) [inline doc/sphinx/installation/dependencies.rst:68]: Include `mkdir <p4est_source>` for completeness?

File: doc/sphinx/installation/dependencies.rst [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+where p4est-x.y.z.tar.gz is the name of the file and x.y.z the version. All stable releases of p4est
+can be found `here <https://github.com/p4est/p4est.github.io/tree/master/release/>`_.
+
+Install p4est making use of a configuration scipt provided by deal.II:
+
+.. code-block:: console
+
+   cd <p4est_source>
+   <dealii_source>/doc/external-libs/p4est-setup.sh p4est-x.y.z.tar.gz <p4est_install>
```
Comment (ppraegla) [inline doc/sphinx/installation/dependencies.rst:80]: Include `mkdir <p4est_install>` for completeness?

File: doc/sphinx/installation/dependencies.rst [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+Configure and install deal.II with p4est:
+
+.. code-block:: console
+
+   cd <dealii_build>
```
Comment (ppraegla) [inline doc/sphinx/installation/dependencies.rst:87]: Include `mkdir <dealii_build>` for completeness?

File: doc/sphinx/installation/configuration.rst [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+
+PartExa library
+~~~~~~~~~~~~~~~
+
+In the following it is assumed that all :ref:`dependencies` are met. Configure and build PartExa
+following
+
+.. code-block:: console
+
+   cd <partexa_build>
```
Comment (ppraegla) [inline doc/sphinx/installation/configuration.rst:46]: Include `mkdir <partexa_build>` for completeness?

File: CMakeLists.txt [hunk 5]
```diff
@@ -2,7 +2,7 @@
 ##
 ## PartExa - A Particle Library for the Exa-Scale
 ##
-## Copyright (C) 2021 by the PartExa authors
+## Copyright (C) 2022 by the PartExa authors
```
Comment (ppraegla) [inline CMakeLists.txt]: I think the copyright notice should be
```suggestion
## Copyright (C) 2021-2022 by the PartExa authors
```

**Review by ppraegla — CHANGES_REQUESTED** [review]
Thanks for getting started with the documentation. I have only some minor remarks.

File: doc/sphinx/installation/configuration.rst (hunk 4 above)
Comment (slfuchs) [reply doc/sphinx/installation/configuration.rst:46]: I'd like to keep the guide brief and would rather add one note above that states that all directory paths embraced in `<...>` need to be created by the user. What do you think?

Comment (ppraegla) [reply doc/sphinx/installation/configuration.rst:46]: Imo adding the mkdir directives wouldn't add too much code and makes it easier to follow the steps.

File: CMakeLists.txt (hunk 5 above)
Comment (slfuchs) [reply CMakeLists.txt]: I changed/reset all copyright years as discussed yesterday.

File: doc/sphinx/installation/configuration.rst (hunk 4 above)
Comment (slfuchs) [reply doc/sphinx/installation/configuration.rst:46]: Added all `mkdir` as discussed yesterday.

**Comment by slfuchs** [comment]
@ppraegla Please take a look and if you are ok with the PR feel free to directly merge.

**Review by ppraegla — APPROVED** [review]
Looks good. Thanks!

**Pull request merged**